crypto.go

```diff
@@ -199,6 +200,35 @@ func blindBaseElement(blindingFactor btcec.ModNScalar) *btcec.PublicKey {
 	return priv.PubKey()
 }
 
+// chacha20polyEncrypt initialises the ChaCha20Poly1305 algorithm with the given
+// key and uses it to encrypt the passed message.
+func chacha20polyEncrypt(key, msg []byte) ([]byte, error) {
+	aead, err := chacha20poly1305.New(key)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := bytes.Repeat([]byte{0x00}, aead.NonceSize())
+
+	// Encrypt the message and append the ciphertext to the nonce.
+	encryptedMsg := aead.Seal(nil, nonce, msg, nil)
+	return encryptedMsg, nil
+}
+
+// chacha20polyDecrypt initialises the ChaCha20Poly1305 algorithm with the given
+// key and uses it to decrypt the passed cipher text.
+func chacha20polyDecrypt(key, cipherTxt []byte) ([]byte, error) {
+	aead, err := chacha20poly1305.New(key)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := bytes.Repeat([]byte{0x00}, aead.NonceSize())
+
+	// Decrypt the message and append the ciphertext to the nonce.
+	return aead.Open(nil, nonce, cipherTxt, nil)
```

Comment:
I _believe_ we can use `cipherTxt[:0]` intead of nil for dst if we want to be storage-thrifty.